datasource/src/main/scala/quasar/plugin/mariadb/datasource/MariaDbDatasourceModule.scala

```diff
@@ -0,0 +1,133 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.mariadb.datasource
+
+import scala._
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration._
+import scala.util.Either
+
+import java.lang.String
+import java.net.URI
+
+import argonaut._, Argonaut._, ArgonautCats._
+
+import cats.data.{NonEmptyList, NonEmptySet}
+import cats.effect._
+import cats.implicits._
+
+import doobie._
+
+import quasar.RateLimiting
+import quasar.api.datasource.{DatasourceType, DatasourceError}
+import quasar.api.datasource.DatasourceError.ConfigurationError
+import quasar.connector.{ByteStore, MonadResourceErr}
+import quasar.connector.datasource.{LightweightDatasourceModule, Reconfiguration}
+import quasar.plugin.jdbc.{JdbcDiscovery, Redacted, TableType, TransactorConfig}
+import quasar.plugin.jdbc.JdbcDriverConfig
+import quasar.plugin.jdbc.datasource.JdbcDatasourceModule
+
+import org.slf4s.Logger
+
+import scalaz.{NonEmptyList => ZNel}
+
+object MariaDbDatasourceModule extends JdbcDatasourceModule[DatasourceConfig] {
+
+  val DefaultConnectionMaxConcurrency: Int = 8
+  val DefaultConnectionMaxLifetime: FiniteDuration = 5.minutes
+
+  val kind = DatasourceType("mariadb", 1L)
+
+  def discoverableTableTypes(log: Logger): Option[ConnectionIO[NonEmptySet[TableType]]] =
+    Some(for {
+      catalog <- HC.getCatalog
+      rs <- HC.getMetaData(FDMD.getTableTypes)
+      names <- FC.embed(rs, HRS.build[SortedSet, String])
+      _ <- FC.delay(log.debug(s"AVAILABLE TABLE TYPES: ${names.toList.mkString(", ")}"))
+      pruned = names.filterNot(n => n == "SYSTEM TABLE" || n == "SYSTEM VIEW")
+      default = NonEmptySet.of("TABLE", "VIEW")
+      discoverable = NonEmptySet.fromSet(pruned) getOrElse default
+    } yield discoverable.map(TableType(_)))
+
+  def transactorConfig(config: DatasourceConfig): Either[NonEmptyList[String], TransactorConfig] =
+    for {
+      cc <- config.connectionConfig.validated.leftMap(NonEmptyList.one(_)).toEither
+
+      jdbcUrl <-
+        Either.catchNonFatal(new URI(cc.jdbcUrl))
+          .leftMap(_ => NonEmptyList.one("JDBC URL is not a valid URI"))
+
+      maxConcurrency = cc.maxConcurrency getOrElse DefaultConnectionMaxConcurrency
+      maxLifetime = cc.maxLifetime getOrElse DefaultConnectionMaxLifetime
+    } yield {
+      TransactorConfig
+        .withDefaultTimeouts(
+          JdbcDriverConfig.JdbcDriverManagerConfig(
+            jdbcUrl,
+            Some("org.mariadb.jdbc.Driver")),
+          connectionMaxConcurrency = maxConcurrency,
+          connectionReadOnly = false)
```

Comment:
Given that we're using the db as a datasource, why can't this be read-only?

Reply:
Um, no reason, good catch 👍